### PR TITLE
Add config management api & implementation

### PR DESCRIPTION
**Pull request by indiketa**
Added full support for config management to docker-java
[Docs](https://docs.docker.com/engine/api/v1.39/#tag/Config)

#1354

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,360 +1,278 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-	<modelVersion>4.0.0</modelVersion>
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (bsideup) [inline pom.xml:2]: could you please revert this unrelated changes?

Comment (indiketa) [reply pom.xml]: The IDE did this without notice, reverting...

Comment (indiketa) [reply pom.xml]: @bsideup  Changes done (i had to do a force-push), waiting your review, thank you.

**Comment by trajano** [comment]
@indiketa can you add support to get data as string?  I think most use cases would store the config data as text.

Something like

```
    /**
     * This gets the decoded data as UTF-8 string.
     *
     * @return decoded data
     */
    public String getDataAsString() {
        return new String(getData(), StandardCharsets.UTF_8);
    }
```

File: docker-java-api/src/main/java/com/github/dockerjava/api/model/ConfigSpec.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    public ConfigSpec withName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    /**
+     * @see #data
+     */
+    public byte[] getData() {
+        return Base64.getDecoder().decode(data);
```
Comment (trajano) [inline docker-java-api/src/main/java/com/github/dockerjava/api/model/ConfigSpec.java:57]: This should really use 
```
Base64.getUrlDecoder().decode(data)
```
The specs specify that it is URL safe

File: docker-java-api/src/main/java/com/github/dockerjava/api/model/ConfigSpec.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    /**
+     * @see #data
+     */
+    public ConfigSpec withData(byte[] data) {
+        this.data = Base64.getEncoder().encodeToString(data);
```
Comment (trajano) [inline docker-java-api/src/main/java/com/github/dockerjava/api/model/ConfigSpec.java:64]: This should really use 
```
Base64.getUrlEncoder().encodeToString(data)
```
The specs specify that it is URL safe

Comment (trajano) [reply docker-java-api/src/main/java/com/github/dockerjava/api/model/ConfigSpec.java:64]: Can you add the following

    /**
     * This gets the decoded data as UTF-8 string.
     *
     * @return decoded data
     */
    public String getDataAsString() {
        return new String(getData(), StandardCharsets.UTF_8);
    }

I think most use cases would store the config data as text.

**Comment by bsideup** [comment]
@indiketa please update with the latest master (note that `SwarmCmdIT` is always using Docker-in-Docker now). Also, make sure that all newly added commands have tests, not just ListConfig. Thanks!
